🚚 Update: Add 104th CatchUp Summary (#112)
OurTechCommunity/catchup: summary/sessions/104/content.adoc

Date: 05-11-2022

Duration: 4 hrs 47 mins

==== Topics Discussed

> * We discussed how college examinations might be starting late for the students in autonomous and colleges under Mumbai University.

Reviewer: Does this add any value?

Author: No, it doesn't.

        ***link:https://krondo.com/an-introduction-to-asynchronous-programming-and-twisted[Asynchronous Programming and the twisted networking framework^] 
        *** link:https://krondo.com/in-which-we-begin-at-the-beginning[In which we begin at the beginning^]
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] and link:https://twitter.com/PranavDani3[Pranav Dani^] talked about the correlation between cores and threads. For instance, a 4-core 8-thread processor has 4 physical cores and each core can run 2 threads in parallel. 
    ** We discussed the difference between Concurrency and Parallelism and why using them interchangeably is wrong.
        *** Concurrency is the composition of independently executing processes.
        *** Parallelism is the simultaneous execution of multiple (possibly related) processes.
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared a video about link:https://youtu.be/_3LpJ6I-tzc[Async I/O in Depth^]
* link:https://twitter.com/WilfredAlmeida_[Wilfred Almeida] talked about problems he was facing with link:https://en.wikipedia.org/wiki/Advanced_Encryption_Standard[AES ciphers^] and its use case in one of his projects. 

>     ** We talked about link:https://meltdownattack.com[Meltdown^] for servers and how these vulnerabilities could be potentially handled.

Reviewer: I don't think there is any correlation between this and above point and these are vulnerabilities are in processors

```suggestion
* We talked about link:https://meltdownattack.com[Meltdown and Spectre^] which are vulnerabilities in modern processors.
	** Meltdown is patched in all the major OS
```

    ** Twitter recently underwent many changes in leadership and as a result, many employees were laid off.
    ** link:https://twitter.com/ashwinexe[Ashwin Kumar Uppala^] and link:https://twitter.com/tusharnankanii[Tushar Nankani^] shared link:https://layoffs.fyi/[layoffs.fyi^] as a resource to find out about layoffs in tech companies.
    ** Decrease in business could be a potential reason for layoffs since it results in a decrease in revenue.
    ** link:https://twitter.com/skxrxn[Sreekaran Srinath^] talked about how stock prices for companies were decreasing in value.
* We talked about some platforms such as link:https://www.microsoft.com/en-in/microsoft-teams/group-chat-software[MS Teams^] and link:https://zoom.us[Zoom^] that gained traction over the pandemic period and now serves as an integral part of the system.
    ** link:https://www.microsoft.com/en-in/microsoft-teams/group-chat-software[MS Teams^] is catered towards enterprises with its tight integration with Office365 tools and provides a relatively better experience in terms of handling bandwidth at even lower tiers. 

>     ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] talked about link:https://dyte.io[Dyte^] and its functionalities while comparing it with link:https://www.100ms.live[100ms^].

Reviewer: Again, I don't feel there is correlation.
```suggestion
* link:https://twitter.com/mehraramyak[Ramyak Mehra^] talked about link:https://dyte.io[Dyte^] and its functionalities while comparing it with link:https://www.100ms.live[100ms^].
```

* link:https://twitter.com/adityaoberai1[Aditya Oberai^] talked about how some layoff panics are a part of the echo chamber, where the layoff news seems to be exaggerated and amplified by multiple people.
    ** Layoffs might hurt less when you are early in your career without many dependencies.
* link:https://twitter.com/SirusTweets[Darshan Rander^] talked about potential reasons behind not choosing a startup, where you do not get any form of guidance or a course correction wherever the candidate might go wrong.

>     ** link:https://twitter.com/adityaoberai1[Aditya Oberai^] talked about the status of an Individual contributor that is allowed by startups which helps with a personal brand.

Reviewer: ```suggestion
    ** link:https://twitter.com/adityaoberai1[Aditya Oberai^] talked about the status of an individual contributor that is allowed by startups which helps with a personal brand.
```

    ** link:https://twitter.com/guyinthecape[Vatsal Patel^] talked about the importance of feedback from seniors and how it might be lacking in startups.

>     ** A company should be selected after weighing all the pros and cons.

Reviewer: There was another POV that if there is not much upper management you could work on your pace and reduce bureaucracy

* link:https://twitter.com/guyinthecape[Vatsal Patel^] shared his experience with releasing link:https://www.callofduty.com/modernwarfare2[Call of Duty: Modern Warfare 2^] at Activision and all the challenges encountered in the process including team management.
* link:https://twitter.com/SirusTweets[Darshan Rander^] shared that he's going to give a talk about "Server Side Dart" at link:https://www.meetup.com/mumbai-women-coders/events/289400920[MWC Meetup^].
    ** He shared his points of benchmark for python and dart backend.

>     ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared that he would like to give a talk about how to approach a new codebase.

Reviewer: ```suggestion
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared link:https://github.com/mcollina/autocannon[HTTP/1.1 benchmarking tool] to better benchmark server in a range of load on it.
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared that he would like to give a talk about how to approach a new codebase.
```

* link:https://twitter.com/kaku_jay[Jay Kaku^] talked about the process behind building his final-year hardware project in which he had to learn about how most basic components work.
* We talked about the ostensible pitty and sad state of the USA's gun laws and how it is a major reason behind the high number of deaths due to gun violence.

> * We discussed how multiple customers tend to compare cars based on vibrations and the kind of sound that it makes which seems redundant, especially when talking about electric cars.

Reviewer: How will this help people? Like will it stop them from going to US? 😜

* link:https://twitter.com/kaku_jay[Jay Kaku^] talked about AMD's latest CPU lineup with a 5nm process and how the graphics technology with RDNA3 competes with Nvidia's RTX 30 and 40 series. 

> * link:https://twitter.com/adityaoberai1[Aditya Oberai^] talked about his setup of cameras and monitors and how he uses his green screen as a background.

Reviewer: I guess we can remove this as well

Date: 05-11-2022

Duration: 4 hrs 47 mins

==== Topics Discussed

* We discussed how college examinations might be starting late for the students in autonomous and colleges under Mumbai University.
* link:https://twitter.com/tusharnankanii[Tushar Nankani^] shared his learnings from an article that pushed him to read more about cores, processors, threads, cache and sockets.
    ** Article links for "Twisted" and the first installment: 
        *** link:https://krondo.com/an-introduction-to-asynchronous-programming-and-twisted[Asynchronous Programming and the Twisted networking framework^] 
        *** link:https://krondo.com/in-which-we-begin-at-the-beginning[In which we begin at the beginning^]
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] and link:https://twitter.com/PranavDani3[Pranav Dani^] talked about the correlation between cores and threads. For instance, a 4-core 8-thread processor has 4 physical cores and each core can run 2 threads in parallel. 

>     ** We discussed the difference between Concurrency and Parallelism and why using them interchangeably is wrong.

Reviewer 2: ```suggestion
    ** We discussed the link:https://twitter.com/tusharnankanii/status/1588639999972761600[difference between Concurrency and Parallelism] and why using them interchangeably is wrong.
```

Adding more context.

        *** Concurrency is the composition of independently executing processes.
        *** Parallelism is the simultaneous execution of multiple (possibly related) processes.
    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared a video about link:https://youtu.be/_3LpJ6I-tzc[Async I/O in Depth^]
* link:https://twitter.com/WilfredAlmeida_[Wilfred Almeida] talked about problems he was facing with link:https://en.wikipedia.org/wiki/Advanced_Encryption_Standard[AES ciphers^] and its use case in one of his projects. 
    ** We talked about link:https://meltdownattack.com[Meltdown^] for servers and how these vulnerabilities could be potentially handled.

> * link:https://twitter.com/tusharnankanii[Tushar Nankani^] talked about the thought process behind writing the tweet regarding his recent project link:https://github.com/tusharnankani/blurrit[blurrit^].

Reviewer 2: ```suggestion
* link:https://twitter.com/tusharnankanii[Tushar Nankani^] talked about the thought process behind writing the link:https://twitter.com/tusharnankanii/status/1579943971648045058[tweet] regarding his recent project link:https://github.com/tusharnankani/blurrit[blurrit^].
```

    ** Layoffs might hurt less when you are early in your career without many dependencies.
* link:https://twitter.com/SirusTweets[Darshan Rander^] talked about potential reasons behind not choosing a startup, where you do not get any form of guidance or a course correction wherever the candidate might go wrong.
    ** link:https://twitter.com/adityaoberai1[Aditya Oberai^] talked about the status of an Individual contributor that is allowed by startups which helps with a personal brand.
    ** link:https://twitter.com/guyinthecape[Vatsal Patel^] talked about the importance of feedback from seniors and how it might be lacking in startups.
    ** A company should be selected after weighing all the pros and cons.
* link:https://twitter.com/guyinthecape[Vatsal Patel^] shared his experience with releasing link:https://www.callofduty.com/modernwarfare2[Call of Duty: Modern Warfare 2^] at Activision and all the challenges encountered in the process including team management.
* link:https://twitter.com/SirusTweets[Darshan Rander^] shared that he's going to give a talk about "Server Side Dart" at link:https://www.meetup.com/mumbai-women-coders/events/289400920[MWC Meetup^].

>     ** He shared his points of benchmark for python and dart backend.

Reviewer 2: ```suggestion
    ** He shared his points of benchmark for Python and Dart backend.
```

Author: yep

    ** link:https://twitter.com/mehraramyak[Ramyak Mehra^] shared that he would like to give a talk about how to approach a new codebase.
* link:https://twitter.com/kaku_jay[Jay Kaku^] talked about the process behind building his final-year hardware project in which he had to learn about how most basic components work.

> * We talked about the ostensible pitty and sad state of the USA's gun laws and how it is a major reason behind the high number of deaths due to gun violence.

Author: not pitty ig

    ** link:https://twitter.com/skxrxn[Sreekaran Srinath^] insisted to keep having fun in the process and focus on problem solving.
* We discussed how we can find some good tech discussions and take value from these perspectives on Twitter. 
    ** Tushar Nankani pointed out how we can leverage Tech Twitter and what we see on our "feed" is not the Tech Twitter. He mentioned that he uses Twitter in a chronological order of the specific people for whom the notifications are turned on. "Doesn't let the algorithm fool you."    
    ** link:https://twitter.com/adityaoberai1[Aditya Oberai^] and link:https://twitter.com/tusharnankanii[Tushar Nankani^] shared a few recommendations for whom the notifications should be turned on: 
        *** https://twitter.com/dan_abramov
        *** https://twitter.com/addyosmani
        *** https://twitter.com/simonw
        *** https://twitter.com/steveruizok
        *** https://twitter.com/MicheBarks
        *** https://twitter.com/vponamariov
        *** https://twitter.com/mhevery
        *** https://twitter.com/Grady_Booch
        *** https://twitter.com/kocienda

>         *** https://twitter.com/matteocollina

Reviewer 2: Thanks, @PranavDani!
